lib/app.js

```diff
@@ -404,17 +405,20 @@ module.exports = {
             return value;
           }
         }
-      }];
-
-      getPrompt(argv, schema, function(results) {
-        if (path.relative('.', results.public).match(/^\./)) {
-          console.log(chalk.red('init cancelled - the public directory must be within the current working directory'));
-          process.exit(1);
+      };
+      schemaPublic.conform = function(value) {
+        if (path.relative('.', value).match(/^\./)) {
+          schemaPublic.message = 'Please use a sub directory of current working directory';
+          return false;
         }
-        if (!fs.existsSync(results.public)) {
-          console.log(chalk.red('init cancelled - the directory you entered does not exist'));
-          process.exit(1);
+        if (!fs.existsSync(value)) {
+          schemaPublic.message = 'Please use an existing directory';
```

Comment:
reasoning: 'Please use an existing directory' doesn't have a clear CTA and implies that the folder has to already exist. There's nothing to stop someone creating a folder in the interim and running the prompt again. I actually wonder if we should create the folder for them here, or at least only check this when actually deploying to Firebase Hosting (they might want to have a build folder that is generated by some script later)